### PR TITLE
Adding per-file editable and lock icons in tabs

### PR DESCRIPTION
**Pull request by TeddyBones**
This adds an editable option for each item in the files and links arrays.
It also adds a lock icon in tabs that are read only.

My first pull request, so I hope I did everything correctly. 😄

### CODE REVIEW

**Comment by porsager** [comment]
Cool 🙂 It looks good, just a few comments.. My first time reviewing a pull request 😉 hehe...

**Comment by TeddyBones** [comment]
Sure, just let me know what to change and I'll change it. :smile:

**Comment by porsager** [comment]
Heh.. Apparently I missed the "submit review" button yesterday, even though I had made the comments then... Well.. You should be able to see them now ;)

File: src/editor/editor.js [hunk 1]
```diff
@@ -158,6 +158,8 @@ export default (model, actions) =>
 
         cm.setOption('lineWrapping', mode.lineWrapping || false)
 
+        cm.setOption('readOnly', (file.editable === undefined) ? false : !file.editable)
```
[the remark this answers is not in the document]
Comment (TeddyBones) [reply src/editor/editor.js]: But only if it's false, right?

File: src/editor/tabs.js [hunk 2]
```diff
@@ -27,11 +27,12 @@ export default (model, actions) =>
 function linkTabs(model, actions) {
   return model.state.links.map(link =>
     tab(
-      m('a' + b.c('inherit'), {
+      [m('a' + b.c('inherit'), {
         href: link.url,
         target: '_blank',
         onclick: e => model.linkContent[link.url] && e.preventDefault()
       }, link.name),
+      m.trust(link.editable === false ? ' 🔒' : '')],
```
[the remark this answers is not in the document]
Comment (TeddyBones) [reply src/editor/tabs.js]: Ah, sorry. Got lost when trying to figure out mithril. 🙂

**Comment by TeddyBones** [comment]
Haha, no problem

File: src/editor/editor.js [hunk 3]
```diff
@@ -156,9 +156,11 @@ export default (model, actions) =>
         const content = file.content || model.linkPatched[file.url] || ''
             , mode = modes[file.name.split('.').pop()] || 'javascript'
 
+        const editable = (model.state.editable === false || file.editable === false) ? false : true
```
Comment (porsager) [inline src/editor/editor.js]: Yeah almost there :P Still should be editable if file.editable is not set, so this should do it
```
const editable = model.state.editable && file.editable !== false
```

Comment (TeddyBones) [reply src/editor/editor.js]: I think it already is, but your code is better, so I'll push that in a minute.

Comment (porsager) [reply src/editor/editor.js]: I'm sorry, you're absolutely right! Fine to keep it as is ;)

**Comment by TeddyBones** [comment]
Awesome! :)

**Comment by porsager** [comment]
I just created an icon matching the others for the lock and did a new npm release with your additions ;)

**Comment by TeddyBones** [comment]
Cool! Nice icons btw.

**Pull request merged**